inventory-consumer/handler.go

```diff
@@ -1,15 +1,52 @@
 package inventoryconsumer
 
 import (
+	"config-manager/domain"
+	"config-manager/domain/message"
+	kafkaUtils "config-manager/infrastructure/kafka"
 	"context"
+	"encoding/json"
 	"fmt"
 
 	kafka "github.com/segmentio/kafka-go"
 )
 
 type handler struct {
+	ConfigManagerService domain.ConfigManagerInterface
 }
 
 func (this *handler) onMessage(ctx context.Context, msg kafka.Message) {
-	fmt.Println("Message: ", msg.Value)
+	eventType, err := kafkaUtils.GetHeader(msg, "event_type")
+	if err != nil {
+		fmt.Println("Error getting header: ", err)
+		return
+	}
+
+	if eventType == "delete" {
+		return
+	}
+
+	value := &message.InventoryEvent{}
+
+	if err := json.Unmarshal(msg.Value, &value); err != nil {
+		fmt.Println("Couldn't unmarshal inventory event: ", err)
+		return
+	}
+
+	if value.Host.Reporter == "cloud-connector" {
+		accState, err := this.ConfigManagerService.GetAccountState(value.Host.Account)
+		if err != nil {
+			fmt.Println("Error retrieving state for account: ", value.Host.Account)
```

Comment:
Do you want to return here in the case where an account state does not exist for an account?  It seems like that could happen in the case where the rhc client has connected to cloud-connector but the customer has not gone into the UI and configured an expected state, right?

If that is a valid use case, then you might add a test for it in your unit tests.  That will help document that its a valid use case.

Reply:
If the account state doesn't exist then a default state is created and used. I believe this is the behavior we'd want to see so that any rhc clients still get insights-client/compliance/etc even without heading to the UI previously. Here's the code: https://github.com/RedHatInsights/config-manager/blob/master/application/config_manager_service.go#L32

And yeah I definitely need some tests around that.

Reply:
Ah, sounds good.  That makes sense.